src/shared/telemetry/activation.ts

```diff
@@ -72,24 +75,28 @@ export async function activate(activateArguments: {
         activateArguments.extensionContext.subscriptions
     )
 }
-
-export function isTelemetryEnabled(toolkitSettings: SettingsConfiguration): boolean {
+/*
+ * Formats the AWS telemetry setting to a boolean: false if setting value was 'Disable' or false, true for everything else
+ */
+export async function formatTelemetrySettingToBool(toolkitSettings: SettingsConfiguration): Promise<void> {
     // Setting used to be an enum, but is now a boolean.
     // We don't have api-based strong type support, so we have to process this value manually.
     const value = toolkitSettings.readSetting<any>(AWS_TELEMETRY_KEY)
 
-    // Handle original opt-out value (setting used to be a tri-state string value)
-    if (value === LEGACY_SETTINGS_TELEMETRY_VALUE_DISABLE) {
-        return false
-    }
+    if (typeof value === 'boolean') return
 
-    // Current value is expected to be a boolean
-    if (typeof value === 'boolean') {
-        return value
+    // Original setting used to be a tri-state string value
+    // Set to false if original setting was opt-out
+    // Treat anything else (unexpected values, datatypes, or undefined) as opt-in
+    if (value === LEGACY_SETTINGS_TELEMETRY_VALUE_DISABLE) {
+        await toolkitSettings.writeSetting<any>(AWS_TELEMETRY_KEY, false, vscode.ConfigurationTarget.Global)
+    } else {
+        await toolkitSettings.writeSetting<any>(AWS_TELEMETRY_KEY, true, vscode.ConfigurationTarget.Global)
```

Comment:
This will unintentionally opt users into telemetry if they've got a typo in their config beyond the 'Enabled' 'Disabled. 

e.g. they have 'Foo' or 123.

I would swap the ordering of these. Check for Enabled, otherwise turn off telemetry. Much safer to assume an opt out.

Reply:
That is not consistent with the rest of the toolkit's handling of telemetry

Reply:
You mean auto opt in? I'd say this edge case situation is different in that there's a value there but it's just invalid. Safer to assume this was an attempt to opt out than to assume it wasn't.